src/reducer.js

```diff
@@ -0,0 +1,54 @@
+import { mapValues } from './utils';
+
+const initialState = {
+  newId: 100,
+  restaurant: {
+    name: '',
+    category: '',
+    address: '',
+  },
+  restaurants: [],
+};
+
+const reducers = {
+  addRestaurant(state) {
+    const { newId, restaurant, restaurants } = state;
+
+    return ({
+      newId: newId + 1,
+      restaurant: {
+        ...mapValues(restaurant, () => ''),
+      },
+      restaurants: [
+        ...restaurants,
+        {
+          id: newId,
+          ...mapValues(restaurant, (value) => value.trim()),
+        },
+      ],
+    });
+  },
+  updateRestaurant(state, action) {
+    const { restaurant } = state;
+    const { field, value } = action.payload;
+
+    return ({
+      ...state,
+      restaurant: {
+        ...restaurant,
+        [field]: value,
+      },
+    });
+  },
```

Comment:
```suggestion
    const { name, value } = action.payload;
    return {
      ...state,
      restaurant: {
        ...state.restaurant,
        [name]: value,
      },
    };
```

보통 field는 좀 더 상위 개념이고

field가 name, value 쌍으로 이루어지죠.